### ISSUE TITLE
Missing column when using read_excel

### ISSUE THREAD

**Issue by reinierv4** [tidyverse/readxl#99]
I try to read in a .xls file that is downloaded from the internet. When I use 

df=read_excel("FILEPATH",col_names=FALSE)  

The df is created but is missing the last column. When I manually open the .xls file first and than click on save this issue disappears and the file is read in normally.To reproduce the behavior it is therefore important to not open the excel file and save "changes" manually before reading it in to R.

An example of the file I try to read in can be downloaded here:

http://www.screencast.com/users/Reinier7840/folders/Default/media/8ab2a43b-6560-4730-890f-e93c1a5ef931

I tried different things like skipping the first line, declaring the columnames and types, and setting col_names equal to true, but that doesn't seem to solve the issue. 

Edit: Loading the same file with the xlsReadWrite package leads to the following error message:

"Error: IOException (Java): block[ 2 ] already removed - does your POIFS have circular or duplicate block references?"

**Comment by drpeterfoster** [comment]
bump.  experienced the same thing.  re-saving .xls fixed the issue.

**Comment by sfirke** [comment]
I have the same problem, caused by some columns having many empty rows before the first value.  E.g., 671 empty rows before first value.

Re-saving did not help, did not matter if it was .xls or .xlsx, it was only reading 117 of 184 variables.  I added a row of dummy values across the whole data set in Excel and then it loaded fine.

**Comment by jomuller** [comment]
I had the same issue with a _.xls_ file. When the last column don't have a value on the row just abowe the header, the column is not imported with `read_excel`. If I add a value (_e.g._ "NA"), the column is correctly imported. To re-save the _.xls_ file with Excel 2010, it also fixed the problem.

**Comment by morsem** [comment]
I'm having a similar issue too. There are 17 columns in the data, but only the first 11 appear. This is because column 12 has empty cells until row 683. My work-around has been just adding place-holders to the first row.

    > library(readxl)
    > 
    > # https://www.dropbox.com/sh/7ldzfux4epgbwfv/AAC6ytAMgO5nt08nyugwwxpVa?dl=0
    > 
    > data <- read_excel("~/Dropbox/readxl_example/de_lose_cols.xlsx", col_names = F)
    > dim(data)
    [1] 7165   11
    > data[1, ]
    Source: local data frame [1 x 11]

                     X0 X1 X2 X3 X4 X5 X6 X7 X8 X9 X10
    1 UNITED STATES SENATOR NA NA NA NA NA NA NA NA NA  NA
    > 
    > data2<- read_excel("~/Dropbox/readxl_example/de_placeholders.xlsx", col_names = F)
    > dim(data2)
    [1] 7165   17
    > data2[1, ]
    Source: local data frame [1 x 17]

                     X0 X1 X2 X3 X4 X5 X6 X7 X8 X9 X10 X11 X12 X13 X14 X15 X16
    1 UNITED STATES SENATOR  X  X  X  X  X  X  X  X  X   X   X   X   X   X   X   X

    > # yes, this data is really messy - here's the first instance of a data in would-be-missing col
    > data2[640:642, ]
    Source: local data frame [3 x 17]

                          X0    X1         X2 X3 X4         X5 X6 X7        X8 X9 X10        X11 X12 X13 X14 X15 X16
    1 REPRESENTATIVE IN CONGRESS    NA         NA NA NA         NA NA NA        NA NA  NA         NA  NA  NA  NA  NA  NA
    2                   District Total   CARNEY J NA NA     IZZO R NA NA  AUGUST B NA  NA    GESTY S  NA  NA  NA  NA  NA
    3                         NA    NA DEMOCRATIC NA NA REPUBLICAN NA NA     GREEN NA  NA LIBERTARIN  NA  NA  NA  NA  NA

**Comment by bjornerstedt** [comment]
Here is an example of the problem. The script test.R opens test.xls file opens correctly with 3 columns with the xlsx package but only with two using readxl. Unfortunately I could not upload the files, so I temporarily put them here:

https://github.com/bjornerstedt/test

**Comment by danielsjf** [comment]
I experienced the same problem. Here is an example file:
https://www.entsoe.eu/fileadmin/template/other/statistical_database/excel.php?pid=137&&opt_period=2&opt_Country=3&opt_Month=1&opt_Year=2008&send=send&opt_Response=99&dataindx=0

The last column never loads.

**Comment by aranryan** [comment]
Thanks drpeterfoster. I was having the same issue, read_excel would miss the last column. Re-saving the Excel file did the trick. It is a pretty ordinary Excel file, no rows with missing data.

**Comment by matteodefelice** [comment]
I had the same problem with the same data by @danielsjf , last column never load. The package xlsx gives me also an error
```
Error in .jcall("RJavaTools", "Ljava/lang/Object;", "invokeMethod", cl,  : 
  org.apache.poi.hssf.OldExcelFormatException: The supplied spreadsheet seems to be Excel 5.0/7.0 (BIFF5) format. POI only supports BIFF8 format (from Excel versions 97/2000/XP/2003)
```
perhaps the files are not proper Microsoft Excel files...

**Comment by petersmp** [comment]
Of note, this appears to still be an issue. I am not sure where the cause is -- these are files generated by a third-party tool as with some of the others (particularly those in other, related, issues). Re-saving (without any other change) worked. Note, I am running Ubuntu 16.04 and using LibreOffice Calc to view the files. The column loss happens whether or not I have opened the file, but goes away once I save.

**Comment by jennybc** [comment]
Once #240 is merged, I expect matters to improve for xlsx (please open a new issue if not!). I'm going to regard this issue as specific to xls going forward.

**Comment by jennybc** [comment]
I can reproduce this and at the moment #271 does not appear to fix it.

**Comment by jennybc** [comment]
Duplicate of #152

**Comment by kamilien1** [comment]
nice catch!

**Comment by ea-guerette** [comment]
Was this issue resolved for xls files? 
I just installed readxl today and am encountering this issue. 
readxl won't import the last column, even though it has a header. 
Opening the file and saving it solves the issue, as mentioned by other users. This is impractical as I have hundreds of files and they keep coming... 
Thanks.

**Comment by jennybc** [comment]
Sounds like you need to open a new issue demonstrating the problems you see.

## How to provide a readxl reprex

We're in a much better position to address your issue if you can provide a reprex (**repr**oducible **ex**ample). Provide as much of this as you can:

  * An actual xls or xlsx file. Pick one:
    - Your personal xls or xlsx: try to strip it down to the minimal size and complexity to demonstrate your point. And, obviously, remove any sensitive data.
    - A publicly available xls or xlsx: provide URL and the code you used to download.
  * A small bit of R code that uses readxl on the provided xls or xlsx file and demonstrates your point.
    - Consider using the [reprex package](https://github.com/jennybc/reprex#readme) to prepare this. In addition to nice formatting, this ensures your reprex is self-contained.
  * Any details about your environment that seem clearly relevant, such as operating system.
        `reprex(..., si = TRUE)`
    will append a standard summary, folded neatly away, at the bottom of your reprex.

How to provide your own xls/xlsx file? In order of preference:

  1. [Attach the file directly to your issue](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/). Instructions are always at the bottom of the issue or comment box.
  1. Share via DropBox or Google Drive and provide the link in your issue.
  1. Explain you absolutely cannot provide a relevant file via github.com and offer to provide privately.
  1. Don't share a file and realize you're hoping for, e.g., a bug fix with no concrete example to go on.

**Comment by ea-guerette** [comment]
Thank you for your fast response.

I have now created an issue (#309). I hope I gave you enough information in my post for you to see what is happening.

Let me know if I have missed something. I am new to this reporting issues thing.

(I usually suffer in silence, but this time I have this urgent project and I need a solution - so thanks for the help!)


Elise

________________________________

**Comment by pranathichunduru** [comment]
Hi ,
I have had the same issue. The R-script read_excel, read_xls,read.xls(from gdata pakcage) only imports say 100 columns of 150 column excel file (50 columns missing) .
I tried re-saving the excel file with dummy values (NA) in the missing rows ,still no luck. 
Is there anything that I am missing or can anyone suggest an alternative.

Thanks

**Comment by jennybc** [comment]
@pranathichunduru Are you absolutely sure you are using the most recent released version of readxl? What exactly is your version number? Use `packageVersion("readxl")`.If you follow the link above, you'll see that a readxl fix did address with that person's problem and sheet.

If this is still a problem with the CRAN version, then I need access to your sheet. It is probably written by some non-Excel tool?

Also, are you missing **rows** or **columns**? Big difference!

You should open a new issue and do this:

## How to provide a readxl reprex

We're in a much better position to address your issue if you can provide a reprex (**repr**oducible **ex**ample). Provide as much of this as you can:

  * An actual xls or xlsx file. Pick one:
    - Your personal xls or xlsx: try to strip it down to the minimal size and complexity to demonstrate your point. And, obviously, remove any sensitive data.
    - A publicly available xls or xlsx: provide URL and the code you used to download.
  * A small bit of R code that uses readxl on the provided xls or xlsx file and demonstrates your point.
    - Consider using the [reprex package](https://github.com/jennybc/reprex#readme) to prepare this. In addition to nice formatting, this ensures your reprex is self-contained.
  * Any details about your environment that seem clearly relevant, such as operating system.
        `reprex(..., si = TRUE)`
    will append a standard summary, folded neatly away, at the bottom of your reprex.

How to provide your own xls/xlsx file? In order of preference:

  1. [Attach the file directly to your issue](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/). Instructions are always at the bottom of the issue or comment box. `.xlsx` is a supported file type. You'll need to zip or gzip `.xls` so it appears as `.zip` or `.gz`.
  1. Share via DropBox or Google Drive and provide the link in your issue.
  1. Explain you absolutely cannot provide a relevant file via github.com and offer to provide privately.
  1. Don't share a file and realize you're hoping for, e.g., a bug fix with no concrete example to go on.

**Comment by SamrawitA** [comment]
Hello, I am using an excel source file which I download from the internet using a script task and the first column doesn't appear in the available input columns, but opening the downloaded file and saving it solves the problem, but the next time the script download's the new file the problem appears again.



below is the link for the sample of the excel file.



Hello, I am using an excel source file which I download from the internet using a script task and the first column doesn't appear in the available input columns, but opening the downloaded file and saving it solves the problem, but the next time the script download's the new file the problem appears again.



below is the link for the sample of the excel file.



http://loraserver.elsys.se/demo/api_v2.php/sensor/A81758FFFE033FF8/all/?no=10
